### PR TITLE
feat: support MLS feature config

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Expose the `IsMLSEnabledUse` use case which returns `true` if MLS has been enabled trough the team settings for the self user.

#### Test Coverage

- [x] I have added automated test to this contribution

#### Notes

The logic for each feature config is currently duplicated in `FeatureConfigEventReceiver.kt` and `SyncFeatureConfigsUseCase.kt`. This needs to be refactored but IMO this outside the scope of this PR.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/Event.kt [hunk 1]
```diff
@@ -84,7 +86,12 @@ sealed class Event(open val id: String) {
     ) : Event(id) {
         data class FileSharingUpdated(
             override val id: String,
-            val status: String
+            val model: ConfigsStatusModel
+        ) : FeatureConfig(id)
+
+        data class MLSUpdated(
+            override val id: String,
+            val model: MLSModel
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/Event.kt:94]: isn't better to have it like MLSConfig? I think MLSModel is not explanatory enough.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/EventMapper.kt [hunk 2]
```diff
@@ -131,7 +133,11 @@ class EventMapper(
     ) = when (featureConfigUpdatedDTO.data) {
         is FeatureConfigData.FileSharing -> Event.FeatureConfig.FileSharingUpdated(
             id,
-            (featureConfigUpdatedDTO.data as FeatureConfigData.FileSharing).status.name
+            featureConfigMapper.fromFeatureConfigsDTO(featureConfigUpdatedDTO.data as FeatureConfigData.FileSharing)
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/EventMapper.kt:136]: I think like other mappers, we can simply name the inner function as 'fromDTO'.

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/client/UserConfigStorage.kt [hunk 3]
```diff
@@ -12,6 +12,16 @@ import kotlinx.serialization.Serializable
 
 interface UserConfigStorage {
 
+    /**
+     * save flag from the user settings to enable and disable the logging
```
Comment (mchenani) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/client/UserConfigStorage.kt:16]: The comment seems to be not correct!

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/client/UserConfigStorage.kt [hunk 4]
```diff
@@ -12,6 +12,16 @@ import kotlinx.serialization.Serializable
...
+     */
+    fun enableMLS(enabled: Boolean)
+
+    /**
+     * get the saved flag to know if the logging enabled or not
```
Comment (mchenani) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/client/UserConfigStorage.kt:21]: here as well.

**Review by mchenani — APPROVED** [review]
✅ Just some comments to consider.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/EventMapper.kt (hunk 2 above)
Comment (typfel) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/EventMapper.kt:136]: 👍  I was just following naming that was already used in the class.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/Event.kt (hunk 1 above)
Comment (typfel) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/Event.kt:94]: Every other feature config is called *Model in the logic module. I'd prefer rename this in separate refactoring PR.

**Pull request merged**